### PR TITLE
Add new unittest and coverage

### PR DESCRIPTION
**Pull request by kaitj91**
This PR adds a new unit test and coverage. This PR is in reference to PR #37 and the suggested changes I received. I now have it so the sqlalchemy engine and session are created and the unittest relies on that rather than flask app. 

I have also added coverage to our tests and updated tox to call both nosetests and py.test.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,4 +6,6 @@ python:
 install:
   - "pip install ."
   - "pip install -r requirements.txt"
-script: py.test
\ No newline at end of file
+script:
+  - py.test /sqlalchemy_jsonapi/tests/
```
Comment (rockstar) [inline .travis.yml]: Hm, this path looks fishy, and it doesn't look like travis has picked up on the tests. I think, at the very least, the path should start with `./`

File: tox.ini [hunk 2]
```diff
@@ -4,9 +4,11 @@ envlist = py27,py3,dandruff
 [testenv]
 use_develop=True
 deps=-r{toxinidir}/requirements.txt
-commands=py.test
+commands=
```
Comment (rockstar) [inline tox.ini]: @deryckh I think this is a fine example for why we should just use tox in travis now. Doesn't make a lot of sense for travis to have one way of running tests, and for us to have another.

File: .travis.yml (hunk 1 above)
Comment (kaitj91) [inline .travis.yml]: Thanks :) I made fix hopefully it will find tests. I completely forgot about ./ since in tox I don't have to worry about that.

**Comment by kaitj91** [comment]
Made necessary changes and answered questions, going to merge.

**Pull request merged**